### DISCUSSION TITLE
How can I return arbitrary json from ariadne?

### DISCUSSION THREAD

**Discussion by shivshankardayal** [mirumee/ariadne#591] [Q&A]
I have a FastAPI aridne app and I need to return arbitrary json from one of the resolvers. How can I achieve that?

**Comment by shivshankardayal** [comment]
I am no longer looking for the answer as I no longer have that project with me but I will leave it for someone who might need the answer.

**Reply by maxmorlocke** [reply]
Can't comment on FastAPI, but on the django side we built a JSON type and set a field to that value.  

https://github.com/reset-button/ariadne_django/blob/main/ariadne_django/scalars/json.py

**Comment by rafalp** [comment]
Assuming that by "return arbitrary JSON" you mean you want result returned by resolver go directly to JSON serializer and skip any other result processing that GraphQL may do, all you need to do is to define custom scalar in schema and use it for your field.

Schema:

```graphql
scalar Generic

type Query {
    test: Generic
}
```

Resolver returning dict:

```python
def resolve_test(*_):
    return {"hello": {"world": ["earth", "mars"]}}
```

GraphQL query:

```
{
  test
}
```

JSON with query's result:

```json
{
  "data": {
    "test": {
      "hello": {
        "world": [
          "earth",
          "mars"
        ]
      }
    }
  }
}
```

I've named scalar "Generic" because thats how it seems to be named most commonly in GraphQL world.

Note that all data returned from resolver has to be JSON-serializable, or Ariadne will crash on response creation.
